_getline.c

```diff
@@ -0,0 +1,52 @@
+#include "shell.h"
+
+/**
+ * _getline - reads an entire line from fd, storing the address of
+ *the buffer containing the text into *lineptr.
+ * @fd: the stream.
+ * @lineptr: the buffer containing the line from fd.
+ * @n: the size of the buffer.
+ *
+ * Return: number of characters read, or -1 on failure.
+ */
+
+ssize_t _getline(char **lineptr, size_t *n, int fd)
+{
+	char *new_ptr;
+	static size_t count;
+
+	if (lineptr == NULL || n == NULL)
+		return (-1); /* error indicating invalid arguments */
+
+	if (*lineptr == NULL || *n == 0)
+	{
+		*n = INIT_SIZE;
+		*lineptr = malloc(*n);
+
+		if (*lineptr == NULL)
+			return (-1); /* error indicating the failure of malloc */
+	}
+
+	while ((count += read(fd, *lineptr + count, INIT_SIZE)) != 0)
```

Comment:
here the count will not be 0, as you always add the reader values to `count` var that may already have a value.